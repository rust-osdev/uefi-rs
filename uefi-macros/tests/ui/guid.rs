use uefi_macros::unsafe_guid;

// The GUID here is OK.
#[unsafe_guid("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa")]
struct Good;

// Fail because the length is wrong.
#[unsafe_guid("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa")]
struct TooShort;

// Error span should point to the second group.
#[unsafe_guid("aaaaaaaa-Gaaa-aaaa-aaaa-aaaaaaaaaaaa")]
struct BadHexGroup2;

// Error span should point to the fifth group.
#[unsafe_guid("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaG")]
struct BadHexGroup5;

fn main() {}
