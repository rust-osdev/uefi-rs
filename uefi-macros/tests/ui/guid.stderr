error: "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa" is not a canonical GUID string (expected 36 bytes, found 35)
 --> $DIR/guid.rs:8:15
  |
8 | #[unsafe_guid("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaa")]
  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: GUID component "Gaaa" is not a hexadecimal number
  --> $DIR/guid.rs:12:25
   |
12 | #[unsafe_guid("aaaaaaaa-Gaaa-aaaa-aaaa-aaaaaaaaaaaa")]
   |                         ^^^^

error: GUID component "aaaaaaaaaaaG" is not a hexadecimal number
  --> $DIR/guid.rs:16:40
   |
16 | #[unsafe_guid("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaG")]
   |                                        ^^^^^^^^^^^^
